x-pack/plugins/asset_manager/server/lib/implicit_collection/collectors/hosts.ts

```diff
@@ -7,12 +7,17 @@
 
 import { APM_INDICES, METRICS_INDICES, LOGS_INDICES } from '../../../constants';
 import { Asset } from '../../../../common/types_api';
-import { CollectorOptions } from '.';
+import { CollectorOptions, QUERY_MAX_SIZE } from '.';
+import { withSpan } from './helpers';
 
-export async function collectHosts({ client, from }: CollectorOptions): Promise<Asset[]> {
+export async function collectHosts({
+  client,
+  from,
+  transaction,
```

Comment:
note: collectors should add pagination labels to the transaction when it is implemented